influxql/ast.go

```diff
@@ -876,6 +906,48 @@ func (s *SelectStatement) Validate(tr targetRequirement) error {
 		}
 	}
 
+	if err := s.validateDerivative(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *SelectStatement) validateDerivative() error {
+	if !s.HasDerivative() {
+		return nil
+	}
+
+	// If a derivative is requested, it must be the only field in the query. We don't support
+	// multiple fields in combination w/ derivaties yet.
+	if len(s.Fields) != 1 {
+		return fmt.Errorf("derivative cannot be used with other fields")
+	}
+
+	aggr := s.FunctionCalls()
+	if len(aggr) != 1 {
+		return fmt.Errorf("derivative cannot be used with other fields")
+	}
+
+	// Derivative requires two arguments
+	derivativeCall := aggr[0]
+	if len(derivativeCall.Args) != 2 {
+		return fmt.Errorf("derivative requires a field and duration argument")
```

Comment:
ideally, duration would be an optional parameter. Where the default if it's a nested aggregate is the same as the group by time. And the default for the raw would be `1s`.